pkgs/os-specific/linux/bionic-prebuilt/default.nix

```diff
@@ -0,0 +1,137 @@
+    sha256 = "0pd8d11md55nd04vgpysgb47ajnnr4kabzdc08g15bvy90a7jsm8";
+    stripRoot = false;
+  };
+
+  prebuilt_ndk_crt = fetchzip {
+    url = "https://android.googlesource.com/platform/prebuilts/ndk/+archive/refs/heads/android11-release/r21/platforms/android-28/arch-arm64/usr/lib.tar.gz";
+    sha256 = "1g7s5gc69x42ryz8mmv4hzqzk985cs009d6s6kbkg550lcrfwvi7";
+    stripRoot = false;
+  };
+
+  ndk_support_headers = fetchzip {
+    url ="https://android.googlesource.com/platform/prebuilts/ndk/+archive/refs/heads/android11-release/r21/sources/cxx-stl/llvm-libc++/include/support.tar.gz";
+    sha256 = "1g7s5gc69x42ryz8mmv4hzqzk985cs009d6s6kbkg550lcrfwvi7";
+    stripRoot = false;
+  };
+
+  kernelHeaders = pkgs.makeLinuxHeaders {
+    arch = "arm64";
+    version = "5.1";
+    src = fetchurl {
+      url =
+        "https://android.googlesource.com/kernel/msm/+archive/db4bfbca77cf68e60e503f84074b853934bce55f.tar.gz";
+      sha256 = "0ki5gh1nsigjdpfpyw9ckbsrska3m94hdq1cxngmwif5rhv1z8vi";
+    };
+  };
+
+in
+stdenvNoCC.mkDerivation rec {
+  pname = "bionic-prebuilt";
+  version = "android11-release";
+
+  src = fetchurl {
+    url = "https://android.googlesource.com/platform/bionic/+archive/refs/heads/${version}.tar.gz";
+    sha256 = "0qklgmpy4x0bq34061d7grzsj0hgw8jbf937fda00a2ivz4l476c";
+  };
+
+  sourceRoot = ".";
+
+  NIX_DONT_SET_RPATH = true;
+
+  dontConfigure = true;
+  dontBuild = true;
+  patchPhase = ''
+    substituteInPlace libc/include/sys/cdefs.h --replace \
+      "__has_builtin(__builtin_umul_overflow)" "1"
+    substituteInPlace libc/include/bits/ioctl.h --replace \
+      "!defined(BIONIC_IOCTL_NO_SIGNEDNESS_OVERLOAD)" "0"
+  '';
+  installPhase= ''
+    # copy the bionic headers
+    echo copying libc files
+    mkdir -p $out/include
+    cp -vr libc/include/* $out/include
+    echo copying kernel headers
+    # copy the kernel headers
+    cp -vr ${kernelHeaders}/include/*  $out/include/
+    substituteInPlace $out/include/asm/sigcontext.h --replace "#ifdef CONFIG_64BIT" "#if 1"
+    # copy the support headers
+    mkdir -p $out/include/support/
+    cp -vr ${ndk_support_headers}* $out/include/support/
+    mkdir -p $out/include/android
+    cat >$out/include/android/ndk-version.h <<END
+#pragma once
+
+/**
+ * Set to 1 if this is an NDK, unset otherwise. See
+ * https://android.googlesource.com/platform/bionic/+/master/docs/defines.md.
+ */
+#define __ANDROID_NDK__ 1
+
+/**
+ * Major version of this NDK.
+ *
+ * For example: 16 for r16.
+ */
+#define __NDK_MAJOR__ 22
+
+/**
+ * Minor version of this NDK.
+ *
+ * For example: 0 for r16 and 1 for r16b.
+ */
+#define __NDK_MINOR__ 0
+
+/**
+ * Set to 0 if this is a release build, or 1 for beta 1,
+ * 2 for beta 2, and so on.
+ */
+#define __NDK_BETA__ 0
+
+/**
+ * Build number for this NDK.
+ *
+ * For a local development build of the NDK, this is -1.
+ */
+#define __NDK_BUILD__ 7026061
+
+/**
+ * Set to 1 if this is a canary build, 0 if not.
+ */
+#define __NDK_CANARY__ 0
+END
```

Comment:
Please convert that into a patch.

Reply:
Yes, and also I believe the `__NDK_MAJOR__` and `__NDK_MINOR__` can be gotten from the `stdenv.hostPlatform.sdkVer`?

Reply:
Those are not the same as the SDK version... they are distinct. I don't believe we use the values anywhere. They just need to be defined.

Reply:
OK

Reply:
Actually, I think it might be still better to put them in Nix (even if they are just `let`-bound with the package and not put in `targetPlatform`) just so people are more likely to remember to update them.